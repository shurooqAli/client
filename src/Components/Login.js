import React, { useEffect, useState } from 'react';
import { Col, Container,FormGroup,Row,Form, Label, Input, Button } from 'reactstrap';
import LOGO from '../Images/logo-t.jpg';
import { UsersData } from '../ExampleData';
import { LoginValidation } from '../Validations/LoginValidation';
import {useForm} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import { Link } from 'react-router-dom';
import {useNavigate} from 'react-router-dom';
import { getUser } from '../Features/UserSlice';
import { useDispatch, useSelector } from 'react-redux';

function Login(){
    let [email,setEmail]=useState("");
    let [password,setPassword]=useState("");
    let message=useSelector((state)=>state.counter.message);
    let user=useSelector((state)=>state.counter.user);
    let isSuccess=useSelector((state)=>state.counter.isSuccess);
    let isError=useSelector((state)=>state.counter.isError);
    let navigate=useNavigate();
    let dispatch=useDispatch();

    const {
        register,
        handleSubmit:submitForm,
        formState:{errors},
    }=useForm({resolver:yupResolver(LoginValidation)});

    const handleSubmit=()=>{
        const user={password:password,email:email};
        dispatch(getUser(user));
    }
    useEffect(()=>{
        if(user&&isSuccess)
            navigate("/BookStore")
        if(isError)
            navigate("/")
    },[isSuccess,isError,user]);

    return(
        <Container fluid>
           <Row className='div-row'>
            <Col className='div-col' md="6">
                <form className='div-form'>
                    <div className='div-logo'>
                        <img src={LOGO} width="200px" height="100px"/>
                        <h2>Book Library</h2>
    
                    </div>
                    <FormGroup>
                        <h6>Login to book library!</h6>
                       
                        <Label>No Account? 
                            <Link to='/register'> Sign Up Now</Link>
                        </Label>

                        </FormGroup>
                    
                    
                    <FormGroup>
                    <Label for="email">Email</Label>
                        <input 
                            type='email'
                            id="email"
                            className='form-control' 
                            {...register("email",{
                                value:email,
                                value:"abc123@gmail.com",
                                onChange:(e)=>setEmail(e.target.value)
                            })} 
                        />
                        <p className='error'>{errors.email?.message}</p>
                    </FormGroup>
                    <FormGroup>
                        <Label>Password</Label>
                        <input 
                            type='password'
                            id="password"  
                            className='form-control' 
                            {...register("password",{
                                value:password,
                                onChange:(e)=>setPassword(e.target.value)
                            })} 
                        />
                        <p className='error'>{errors.password?.message}</p>
                    </FormGroup>
                    <FormGroup>
                        <Input type='checkbox' className='form-control'/>
                        <Label>Remember Me</Label>
                    </FormGroup>
                    <FormGroup>
                        <Label>Forgot Password !</Label>
                    </FormGroup>
                    <FormGroup>
                        <Button color="primary" className='form-control' onClick={submitForm(handleSubmit)}>SIGN IN</Button>
                    </FormGroup>
                    
                </form>
            </Col>
           </Row>
        </Container>
    );
}
export default Login;