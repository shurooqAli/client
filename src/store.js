import { configureStore } from "@reduxjs/toolkit";
import UserReducer from "./Features/UserSlice";
import PostReducer from "./Features/PostSlice";

export const store=configureStore({
    reducer:{counter:UserReducer,
            posts:PostReducer
    }
});