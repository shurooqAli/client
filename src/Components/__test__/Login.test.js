import Login from '../Login';
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from "../../store";
import reducer from '../../Features/UserSlice';
 
test("Match the UI", () => {
    const { asFragment } = render(
        <Provider store={store}>
            <Router>
                <Login />
            </Router>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});
 
const testInitValue = {
    user: {},
    message: "",
    isLoading: false,
    isSuccess: false,
    isError: false
};
 
test("Checking the State", () => {
    const initialState = reducer(undefined, { type: undefined });
    expect(initialState).toEqual(testInitValue);
});
 