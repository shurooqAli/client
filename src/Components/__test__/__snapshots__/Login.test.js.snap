// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the UI 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="div-row row"
    >
      <div
        class="div-col col-md-6"
      >
        <form
          class="div-form"
        >
          <div
            class="div-logo"
          >
            <img
              height="100px"
              src="logo-t.jpg"
              width="200px"
            />
            <h2>
              Book Library
            </h2>
          </div>
          <div
            class="mb-3"
          >
            <h6>
              Login to book library!
            </h6>
            <label
              class="form-label"
            >
              No Account?
              <a
                href="/register"
              >
                 Sign Up Now
              </a>
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              type="email"
            />
            <p
              class="error"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              type="password"
            />
            <p
              class="error"
            />
          </div>
          <div
            class="mb-3"
          >
            <input
              class="form-control form-check-input"
              type="checkbox"
            />
            <label
              class="form-label"
            >
              Remember Me
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Forgot Password !
            </label>
          </div>
          <div
            class="mb-3"
          >
            <button
              class="form-control btn btn-primary"
              type="button"
            >
              SIGN IN
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
